src/ape_ethereum/ecosystem.py

```diff
@@ -338,14 +338,46 @@ def decode_block(self, data: Dict) -> BlockAPI:
 
         return Block.parse_obj(data)
 
+    def _python_type_for_abi_type(self, abi_type: ABIType) -> Any[Type[Any]]:
+        # NOTE: An array can be an array of tuples, so we start with an array check
+        if abi_type.type.endswith("]"):
+            # remove one layer of the potential onion of array
+            new_type = "[".join(str(abi_type.type).split("[")[:-1])
```

Comment:
This is the code that I think would be better to fix in `ethpm_types.abi.ABIType` by adding a method to fetch the inner array type

Reply:
Do itttttt